scripts/documentation/package-schemas/generatePackageSchemas.ts

```diff
@@ -37,8 +37,9 @@ export function generatePackageSchemas(): Promise<void[]> {
   console.log(`${chalk.blue('i')} Generating Package Schemas`);
   const absoluteRoot = resolve(join(__dirname, '../../../'));
 
-  const packages = getPackageMetadataList(absoluteRoot, 'packages', 'docs').map(
-    (packageMetadata) => {
+  const packages = getPackageMetadataList(absoluteRoot, 'packages', 'docs')
```

Comment:
It's not the responsibility of `generatePackageSchemas` to know that `getPackageMetadataList` might return `null` values in the array. Generally, sparse arrays are hard to work with because you don't expect items to be undefined or null.

I could filter out the empty items in `getPackageMetadataList` instead (might the easier using `reduce` or a for loop).

The other option is to use a visitor pattern, so instead of `getPackageMetadataList` we can use something like:

```
findPackageMetadataList(absoluteRoot, 'packages', 'docs', (packageMetadata) => {
   // ...
});
```

Although in this instance I think just filtering out the values from the list is fine without changing the function signature.

Reply:
done. update the name to findPackageMetadataList. i move the filter(Boolean) to inside function findPackageMetadataList